packages/common-test-utils/src/index.ts

```diff
@@ -810,3 +810,104 @@ export class NoteTestPresetsV2 {
     ];
   }
 }
+
+/**
+ * Utility class to make the tests against vault to be concise while having descriptive
+ * failure messages. */
+export class VaultAsserter {
+  private readonly workspaceVault: WorkspaceVault;
+
+  constructor(workspaceVault: WorkspaceVault) {
+    this.workspaceVault = workspaceVault;
+  }
+
+  /** Asserts that no match is found  */
+  async noMatch(noMatchList: string[]) {
+    const hasNoneMatched = await EngineTestUtilsV4.checkVault({
+      ...this.workspaceVault,
+      nomatch: noMatchList,
+    });
+
+    assert(
+      hasNoneMatched,
+      `Found unexpected match in '${JSON.stringify(
+        noMatchList
+      )}' while checking ${this.vaultToString()}`
+    );
+  }
+
+  /** Asserts that match is found  */
+  async match(matchList: string[]) {
+    const hasFoundMatch = await EngineTestUtilsV4.checkVault({
+      ...this.workspaceVault,
+      match: matchList,
+    });
+
+    assert(
+      hasFoundMatch,
+      `Did not find match for '${JSON.stringify(
+        matchList
+      )}' while checking ${this.vaultToString()}`
+    );
+  }
+
+  private vaultToString() {
+    return `Vault fsPath:'${this.workspaceVault.vault.fsPath}' vaultName: '${this.workspaceVault.vault.name}'`;
+  }
+}
+
+/** Asserter utility to have more concise tests, while having descriptive failure messages.
+ *  Supports fluent API. */
```

Comment:
I'm a bit torn with the decision of introducing a fluent interface here.

As I understand, this allows methods to be chained and read like plain english which makes code more readable.
The downside I see here is that the classes can get very large, and would get larger with further attempts to extend what it does. (which in turn can slow us down).

I am aware some of the utility methods we currently have doesn't quite roll off the tongue, but I think we are generally doing a good job with having a descriptive method name (at least the ones that are being added recently), which does a fairly good job at making code readable.

My question is, is the overhead worth the potential readability gain? IMO, code can be readable regardless of how it is structured as long as the developers agree with a consistent convention. 

My opinion is that if we want more clean code, we should spend time discussing and standardizing conventions and cleaning up existing code that doesn't fit what we agreed upon. My concern is that we are exchanging a slight readability increase on the client side (which is currently only us, as this code isn't a library that is exposed for general use)and sacrificing the ease of incremental improvement and timely iteration.

Would love other's opinion on this. Correct me if I'm not understanding this correctly.

Reply:
I realize this may be an effort to introduce DSL like tests (i.e. `vault().should.not.have.note("foo")`)
Is this the goal here? I am not aware of any discussions about this happening so I may be commenting on things that have already been talked about 😅  Sorry if this is something already discussed

Reply:
Let's remove the comment about the fluent API to avoid this turning into something bigger than it is (some wrappers to have descriptive failure messages). While having the client code that uses these wrappers be fairly concise.

Reply:
To follow with @hikchoi, we currently don't use fluent style when writing tests so lets stick with existing conventions

Reply:
> My opinion is that if we want more clean code, we should spend time discussing and standardizing conventions and cleaning up existing code that doesn't fit what we agreed upon. My concern is that we are exchanging a slight readability increase on the client side (which is currently only us, as this code isn't a library that is exposed for general use)and sacrificing the ease of incremental improvement and timely iteration.

Let's push away from any `fluent apis` (which I shouldn't have added in the comment, I was just planning on returning `this` of the asserter not going for any complex full on fluent api) and just discuss using more clear wrappers. Which can actually speed us up rather slow down our iterations. Given that, we end up with less code in tests, clear failure messages which speed up troubleshooting times. 

Maybe it's just me, but it looks like quite a bit of readability improvement of what to assert is trying to achieve:  

```
vaultNoteAsserter(vaultFrom).noMatchByFName("foo");
```

VS

```
expect(
          _.isUndefined(
            NoteUtils.getNoteByFnameV5({
              fname: "foo",
              notes,
              vault: vaultFrom,
              wsRoot,
            })
          )
        ).toBeTruthy();
```

Reply:
Thanks for the clarifications. Let's make some time to discuss this :) I agree that test errors could have better messages than what we have right now.